lib/update/cluster/plan.go

```diff
@@ -475,6 +483,35 @@ func getEtcdVersion(locator loc.Locator, packageService pack.PackageService) (*s
 		locator, searchLabel)
 }
 
+// getCurrentEtcdVersion attempts to read the version of etcd from the etcd version
+// file. This assumes that all cluster (master) nodes run the same version of etcd
+// although this might be invalidated with the use of custom planet image.
+// This does not make sense to have this descripancy though and it should be flagged
+// elsewhere - i.e. at 'tele build' time
+func getCurrentEtcdVersion(stateDir string) (version *semver.Version, err error) {
+	// https://github.com/gravitational/planet/blob/version/7.0.x/tool/planet/constants.go#L87
+	const etcdVersionKey = "PLANET_ETCD_VERSION"
+	f, err := os.Open(filepath.Join(state.EtcdDir(stateDir), "etcd-version.txt"))
```

Comment:
I'm curious, instead of relying on this file being present, why not just do `etcd --version` (or a respective API call)?

Reply:
This an excellent point - I've addressed it in https://github.com/gravitational/gravity/pull/1982/commits/7c19daac69f80530b0eaaf888083aa267ba14593